### ISSUE TITLE
Rust spatial index writing/generation issue

### ISSUE THREAD

**Issue by jcary741** [flatgeobuf/flatgeobuf#192]
I'm running into an issue with flatgeobuf where geometries aren't visible in QGIS. I *think* it's an issue with the rust implementation of the spatial index generation or writing, as skipping writing the spatial index via `header_args.index_node_size = 0` causes the geometries to become visible in QGIS.

## Versions
Rust toolchain:
nightly-x86_64-unknown-linux-gnu
rustc 1.61.0-nightly (458262b13 2022-03-09)
QGIS: 3.22.3
GDAL: 3.4.1

## Replication
1. Uncomment flatgeobuf/src/rust/tests/write.rs:125
2. Comment out the following line (126).
3. Run `cargo test json_to_fgb`
4. Add the resulting file (`test_multipoly.fgb`) to QGIS. No geometries appear. Note that you can view the attributes table, zoom to layer, and selecting all features then copying to a new scratch layer causes them to appear. Therefore, the geometries and attributes are being written correctly.
5. Add `header_args.index_node_size = 0` after line 100, and rename the output file to `test_multipoly2.fgb`
6. Run `cargo test json_to_fgb`
7. Add the resulting file (`test_multipoly2.fgb`) to QGIS. Note that geometries now appear.

## Could this be a GDAL/QGIS issue?
Maybe, but if you run `ogr2ogr -f FlatGeobuf test_multipoly3.fgb test_multipoly.fgb`, which should re-build the spatial index (the default according to [this doc](https://gdal.org/drivers/vector/flatgeobuf.html), the issue is also resolved. So, I really think this is an issue with the rust implementation of the spatial index.

## But what's causing it?
No idea, I've followed this as far I can. I suppose by generating two identical fgb files, one via the c++ flatgeobuf implementation and one via the rust implementation, it might be possible to find where they differ. However, I don't really know, and I won't pretend to understand how packed Hilbert R-Trees work.

**Comment by bjornharrtell** [comment]
Thank for the report @jcary741, it does seem likely there is something wrong in the Rust index generation code.

**Comment by pka** [comment]
I'm now able to reproduce this. When I export the Layer from QGIS, I get a working FGB. `ogrinfo -al` gives practially the same output for both! But the header look quite different in a binary viewer.

@bjornharrtell: Would a `dump-header` and maybe `dump-index` command in fgbutil make sense for you?

**Comment by pka** [comment]
All bounding boxes in the index were empty (Fixed in #207)!

Dumping header and index are in fact one-liners:
```
dbg!(&fgb.header());
fgb.process_index(&mut GeoJsonWriter::new(&mut std::io::stdout()))?;
```

**Comment by bjornharrtell** [comment]
> I'm now able to reproduce this. When I export the Layer from QGIS, I get a working FGB. `ogrinfo -al` gives practially the same output for both! But the header look quite different in a binary viewer.
> 
> @bjornharrtell: Would a `dump-header` and maybe `dump-index` command in fgbutil make sense for you?

Yep!
